python-backend/app/api/search/search/resources/search.py

```diff
@@ -0,0 +1,132 @@
+import regex
+from multiprocessing import Process
+from concurrent.futures import ThreadPoolExecutor, as_completed, wait
+from flask_restplus import Resource, reqparse
+from datetime import datetime
+from flask import request, current_app
+from sqlalchemy import desc, func
+from flask_restplus import fields
+
+from app.extensions import db, api
+from app.api.utils.access_decorators import requires_role_mine_view, requires_role_mine_create
+from app.api.utils.resources_mixins import UserMixin, ErrorMixin
+
+from app.api.mines.mine.models.mine import Mine
+from app.api.parties.party.models.party import Party
+from app.api.permits.permit.models.permit import Permit
+from app.api.documents.mines.models.mine_document import MineDocument
+from app.api.permits.permit_amendment.models.permit_amendment_document import PermitAmendmentDocument
+from app.api.utils.access_decorators import requires_role_mine_view
+from app.api.utils.search import search_targets, SearchResult
+from app.api.search.search_api_models import SEARCH_RESULT_RETURN_MODEL
+
+
+class SearchOptionsResource(Resource, UserMixin):
+    @requires_role_mine_view
+    def get(self):
+        options = []
+        for type, type_config in search_targets.items():
+            options.append({'model_id': type, 'description': type_config['description']})
+
+        return options
+
+
+class SearchResource(Resource, UserMixin):
+    @requires_role_mine_view
+    @api.marshal_with(SEARCH_RESULT_RETURN_MODEL, 200)
+    def get(self):
+        search_results = []
+        app = current_app._get_current_object()
+
+        search_term = request.args.get('search_term', None, type=str)
+        search_types = request.args.get('search_types', None, type=str)
+        search_types = search_types.split(',') if search_types else search_targets.keys()
+
+        reg_exp = regex.compile(r'\'.*?\' | ".*?" | \S+ ', regex.VERBOSE)
```

Comment:
comment or helpful name on what this regex is supposed to do.

Reply:
Added